src/easylink/step.py

```diff
@@ -32,16 +32,16 @@ def __init__(
         self.input_slots = {slot.name: slot for slot in input_slots}
         self.output_slots = {slot.name: slot for slot in output_slots}
         self.parent_step = None
+        self._config = None
 
-    def set_parent_step(self, step):
-        self.parent_step = step
+    @property
+    def config(self) -> LayeredConfigTree:
+        if self._config is None:
```

Comment:
Arent you setting this to None?

Reply:
at the beginning, but we're expecting that no external class will call the `config` property unless something has called `set_step_config` at some point prior